src/js/vic-v2/components/PhotoField.jsx

```diff
@@ -126,26 +119,48 @@ export default class PhotoField extends React.Component {
     }
   }
 
+  onError = (error) => {
+    this.setState({ errorMessage: error });
+  }
+
   onEdit = () => {
-    this.setState({ done: false, warningMessage: null });
+    this.setState({ progress: 0, warningMessage: null });
   }
 
   onDone = () => {
-    const cropResult = this.refs.cropper.getCroppedCanvas().toDataURL(this.state.fileType);
-    this.setState({ cropResult, done: true, warningMessage: null });
+    const filePath = this.props.idSchema.$id.split('_').slice(1);
+    this.refs.cropper.getCroppedCanvas().toBlob(blob => {
+      const file = blob;
+      file.lastModifiedDate = new Date();
+      file.name = this.state.fileName;
+      this.props.formContext.uploadFile(
```

Comment:
The file constructor is not supported in IE or Edge, is this an appropriate format to send, or should we use a [polyfill](https://github.com/moxiecode/moxie) @lihanli @jbalboni ?

Reply:
If that uploads successfully, then yeah, I think that's fine.

Judging by https://stackoverflow.com/questions/33821631/alternative-for-file-constructor-for-safari, what you're doing should work.